buildkite/resource_pipeline_schedule_test.go

```diff
@@ -0,0 +1,201 @@
+package buildkite
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// Confirm that we can add a new pipeline schedule to a pipeline
+func TestAccPipelineSchedule_add_remove(t *testing.T) {
+	var resourcePipeline PipelineNode
+	var resourceSchedule PipelineScheduleNode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckPipelineScheduleResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPipelineScheduleConfigBasic("foo", "0 * * * *"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Schedules need a pipeline
+					testAccCheckPipelineExists("buildkite_pipeline.foobar", &resourcePipeline),
```

Comment:
do we need to check the pipeline in these tests, or can we assume it works thanks to the specific tests for the pipeline resource?

Reply:
Ahh, yeah this is the rub; we don't really need to know that it exists, but if we want to validate we put a schedule on the correct pipeline, we memref it to `testAccCheckPipelineExists` which populates the (until now nil) value of `resourcePipeline`

Reply:
oooohhh, of course. I keep forgetting this methods are also about grabbing an in-memory copy of the data :+1: